### PR TITLE
Quarkus Amazon S3 integration proof of concept

### PR DESCRIPTION
**Pull request by scrocquesel**
Relates to #6705

This a proof of concept of how it could be possible to seamlessly integrate with the quarkus amazon extension.

### CODE REVIEW

File: extensions/aws2-s3/runtime/pom.xml [hunk 1]
```diff
@@ -59,6 +70,13 @@
             <groupId>software.amazon.awssdk.crt</groupId>
             <artifactId>aws-crt</artifactId>
```
Comment (scrocquesel) [inline extensions/aws2-s3/runtime/pom.xml:71]: this should not be necessary, we provide graalvm substitutions https://github.com/quarkiverse/quarkus-amazon-services/blob/main/s3/runtime/src/main/java/io/quarkiverse/amazon/s3/runtime/S3CrtSubstitutions.java

Could be made available in a separate artifact for shared usage

File: extensions/aws2-s3/deployment-quarkus-amazon-s3-integration/src/main/java/org/apache/camel/quarkus/component/aws2/s3/deployment/QuarkusAmazonS3IntegrationProcessor.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+public class QuarkusAmazonS3IntegrationProcessor {
+
+    private static final DotName S3_CLIENT = DotName.createSimple(S3Client.class);
+    private static final DotName S3_PRESIGNER = DotName.createSimple(S3Presigner.class);
+
+    @BuildStep
+    void integrate(BuildProducer<RequireAmazonClientInjectionBuildItem> requireClientInjectionProducer) {
+
+        // request a default bean for S3Client and S3Presigner.
+        requireClientInjectionProducer.produce(new RequireAmazonClientInjectionBuildItem(
```
Comment (scrocquesel) [inline extensions/aws2-s3/deployment-quarkus-amazon-s3-integration/src/main/java/org/apache/camel/quarkus/component/aws2/s3/deployment/QuarkusAmazonS3IntegrationProcessor.java:36]: I can move the build item to a dedicated spi artifact to reduce coupling

**Comment by ppalaga** [comment]
@scrocquesel thanks for the PR. 

Is my understanding correct that he goal is to improve the user experience for applications depending on both Camel Quarkus 
 and Quarkus Amazon? With this change in place, Camel Quarkus would further work without depending on Quarkus Amazon, right?

**Comment by scrocquesel** [comment]
> @scrocquesel thanks for the PR. 
> 
> Is my understanding correct that he goal is to improve the user experience for applications depending on both Camel Quarkus 
>  and Quarkus Amazon? With this change in place, Camel Quarkus would further work without depending on Quarkus Amazon, right?

Right. That's the idea.

**Comment by scrocquesel** [comment]
> The question is whether this would be worth the effort. I was actually considering removing the Quarkus Amazon references & tests from the project. It's mostly a legacy from the early days of the project.
> 
> The vast majority of our users will be using the vanilla Camel component extensions as-is.

Oh great, this would simplify a lot.

We are left with the issue of referencing different versions of the AWS SDK. This could be problematic for the Quarkus Platform, as Quarkus Amazon tends to stay more up to date, and certain versions of the AWS SDK (like the recent 2.29.0) could break native compilation. I know you are copy/pasting some graalvm substitution and maybe this could be improved in some way ?

**Comment by jamesnetherton** [comment]
> We are left with the issue of referencing different versions of the AWS SDK

We could leave the `quarkus-amazon-services-bom` import in our test BOM and that'd ensure we are testing with whatever version of the AWS SDK is used in the platform.

> Quarkus Amazon tends to stay more up to date

True. However, Camel is upgrading to the latest AWS SDK via dependabot (pretty much daily upgrades). So things hopefully wont get too far behind to the extent we have major compatibilty issues.

**Pull request closed without merging**